editors/sc-ide/widgets/code_editor/editor.cpp

```diff
@@ -321,81 +328,93 @@ int GenericCodeEditor::findAll(const QRegExp& expr, QTextDocument::FindFlags opt
+            QString num_str = backref_match.captured(2);
             num_str.remove(0, 1);
             int num = num_str.toInt();
-            // qDebug("%i (%s): backref = %i", i, CSTR(rexpr.cap(2)), num);
-            if (num <= expr.captureCount()) {
-                QString cap = expr.cap(num);
+            // qDebug("%lli (%s): backref = %i", i, CSTR(backref_match.captured(2)), num);
+            if (num <= match.lastCapturedIndex()) {
+                QString cap = match.captured(num);
                 // qDebug("resolving ref to: %s", CSTR(cap));
                 str.replace(i, len, cap);
                 i += cap.size();
             } else {
-                // qDebug("ref out of range", i, num);
+                // qDebug("%lli ref out of range: backref = %i", i, num);
                 str.remove(i, len);
             }
         } else {
-            // qDebug("%i (%s): unknown match", i, CSTR(rexpr.cap(0)));
+            // qDebug("%lli (%s): unknown match", i, CSTR(backref_match.captured(0)));
             str.remove(i, len);
         }
-        // qDebug(">> [%s] %i", CSTR(str), i);
+        // qDebug(">> [%s] %lli", CSTR(str), i);
     }
     // qDebug("END");
     return str;
 }
 
-bool GenericCodeEditor::replace(const QRegExp& expr, const QString& replacement, QTextDocument::FindFlags options) {
-    if (expr.isEmpty())
+bool GenericCodeEditor::replace(const QRegularExpression& expr, const QString& replacement,
+                                QTextDocument::FindFlags options) {
+    if (expr.pattern().isEmpty())
         return true;
 
     QTextCursor cursor = textCursor();
-    if (cursor.hasSelection() && expr.exactMatch(cursor.selectedText())) {
-        QString rstr = replacement;
-        if (expr.patternSyntax() != QRegExp::FixedString)
-            rstr = resolvedReplacement(rstr, expr);
-        cursor.insertText(rstr);
+    bool caps = (expr.patternOptions() & QRegularExpression::PatternOption::DontCaptureOption) == 0;
+    if (cursor.hasSelection()) {
+        QRegularExpressionMatch match = expr.match(cursor.selectedText());
+        if (match.hasMatch()) {
+            QString rstr = replacement;
+            if (caps) {
+                rstr = resolvedReplacement(rstr, match);
+            }
+            cursor.insertText(rstr);
+        }
     }
 
     return find(expr, options);
 }
 
-int GenericCodeEditor::replaceAll(const QRegExp& expr, const QString& replacement, QTextDocument::FindFlags options) {
+int GenericCodeEditor::replaceAll(const QRegularExpression& expr, const QString& replacement,
+                                  QTextDocument::FindFlags options) {
     mSearchSelections.clear();
     updateExtraSelections();
 
-    if (expr.isEmpty())
+    if (expr.pattern().isEmpty())
         return 0;
 
     int replacements = 0;
-    bool caps = expr.patternSyntax() != QRegExp::FixedString;
+    bool caps = (expr.patternOptions() & QRegularExpression::PatternOption::DontCaptureOption) == 0;
```

Comment:
Same here, could this be made a little more explicit?

Reply:
Agree, the variable name also confuses me when i try to reason the original code.